tools/cooja/apps/avrora/build.gradle

```diff
@@ -0,0 +1,24 @@
+apply plugin: "java"
+
+jar.baseName = 'cooja_avrora'
+
+repositories {
+  jcenter()
+}
+
+sourceSets {
+  main {
+    java {
+      srcDir 'src'
+    }
+  }
+
+}
+
+
+dependencies {
+    compile project(":")
+    compile files("lib/avrora-small.jar")
```

Comment:
I understand your concern better now. And I want to say that you would definitely benefit from the setup I suggest (even more than I thought before).

I will try to get some time on the weekend or something to study your build system in more detail. Hope I can help @sieben setup a really great and easy-to-understand Gradle build. Trust me, it can be really great!

Responding to some of your concerns:

* About not seeing the latest jars: you can tell Gradle to always reload the jars from the repo even if their versions did not change (so your build always see the latest jar). See this [discussion](https://discuss.gradle.org/t/how-to-get-gradle-to-download-newer-snapshots-to-gradle-cache-when-using-an-ivy-repository/7344).
* About the repo path: the important, in my view, is not where your repo is going to be located. The important is that projects refer to artifact coordinates only, not jars. Their exact locations do not matter for the projects that depend on it, why should they "know" it? They just tell Gradle, I need this artifact with this group:name:version (version is even optional). I don't care where you'll find. That's a different concern and its solution is specified in the repositories section - that's where you'll worry about where Gradle should look up things. And this is the greatest thing about Gradle. You can use JCenter, Maven Central, your local Maven repo, your file system, even a random url! Have a look at the dependency management/[repositories](https://docs.gradle.org/current/userguide/artifact_dependencies_tutorial.html#N10621) documentation. It's beautiful.

Believe me, there's a huge advantage in separating the concerns of specifying which artifacts you need and specifying where they might be physically located.